ios/MaasCore/Sources/MaasComponents/Components/BottomSheet/BottomSheetPreferenceKeys.swift

```diff
@@ -0,0 +1,26 @@
+import SwiftUI
+
+extension BottomSheet {
+
+    struct ContentPreferenceKey: PreferenceKey {
+        static var defaultValue: CGRect = .zero
+        static func reduce(value: inout CGRect, nextValue: () -> CGRect) { defaultValue = nextValue() }
+    }
+
+    struct PresenterContentPreferenceKey: PreferenceKey {
+        static var defaultValue: CGRect = .zero
+        static func reduce(value: inout CGRect, nextValue: () -> CGRect) { defaultValue = nextValue() }
+    }
+}
+
+extension View {
+
+    func bounds<Key: PreferenceKey>(key: Key.Type, output: @escaping (Key.Value) -> ()) -> some View where Key.Value == CGRect {
```

Comment:
not sure if `bounds` is the best name 🤔 
do you have more options?

Reply:
it could be `contentChanges(key: ..)` because we're observing view content changes. 🤔